src/profile/components/multiItemEditor/__tests__/MultiItemEditor.test.tsx

```diff
@@ -0,0 +1,362 @@
+    },
+    phones: {
+      fields: ['phone'],
+      formValues: {
+        phone: '555-555-123',
+      },
+    },
+    emails: {
+      fields: ['email'],
+      formValues: {
+        email: 'test-email@email.com',
+      },
+    },
+  };
+
+  beforeEach(() => {
+    responses.length = 0;
+  });
+  afterEach(() => {
+    cleanComponentMocks();
+  });
+
+  const initTests = async (dataType: EditDataType): Promise<TestTools> => {
+    responses.push({ profileData: initialProfile });
+    const testTools = await renderTestSuite(dataType);
+    await testTools.fetch();
+    return Promise.resolve(testTools);
+  };
+
+  const multiItemDataTypes: EditDataType[] = ['emails', 'phones', 'addresses'];
+  multiItemDataTypes.forEach(dataType => {
+    // test only node at index 0;
+    const testIndex = 0;
+    const editButtonSelector: ElementSelector = {
+      id: `${dataType}-${testIndex}-edit-button`,
+    };
+    const submitButtonSelector: ElementSelector = {
+      testId: `${dataType}-${testIndex}-save-button`,
+    };
+    const cancelButtonSelector: ElementSelector = {
+      testId: `${dataType}-${testIndex}-cancel-button`,
+    };
+    const addButtonSelector: ElementSelector = {
+      id: `${dataType}-add-button`,
+    };
+
+    const { fields, formValues } = testData[dataType];
+    const sourceNodeList = pickSources(
+      initialProfile,
+      dataType
+    ) as MultiItemProfileNode[];
+    const sourceNode = (sourceNodeList[testIndex] as unknown) as GeneralData;
+
+    // verifies rendered data
+    const verifyValues = async (
+      getTextOrInputValue: TestTools['getTextOrInputValue'],
+      source: GeneralData,
+      index: number,
+      targetIsInput = false
+    ) => {
+      const getSelector = (name: string): GeneralData => {
+        if (dataType === 'addresses') {
+          return targetIsInput
+            ? { id: `${dataType}-${index}-${name}` }
+            : { testId: `${dataType}-${index}-${name}-value` };
+        }
+        return targetIsInput
+          ? { id: `${dataType}-${index}-value` }
+          : { testId: `${dataType}-${index}-value` };
+      };
+      // cannot use forEach with async/await
+      for (const field of fields) {
+        const selector = getSelector(field);
+        const value = source[field];
+        await expect(getTextOrInputValue(selector)).resolves.toBe(value);
+      }
+    };
+
+    // sets new data to input fields
+    const setValues = async (
+      setInputValue: TestTools['setInputValue'],
+      source: GeneralData,
+      index: number
+    ) => {
+      // cannot use forEach with async/await
+      for (const field of fields) {
+        const value = source[field];
+        await setInputValue({
+          selector: {
+            id: `${dataType}-${index}-${
+              dataType === 'addresses' ? field : 'value'
+            }`,
+          },
+          newValue: value,
+        });
+      }
+    };
+
+    describe(`Renders ${dataType} and fields of that 
```

Comment:
I wonder if the language could be improved here hmm.. _Renders_ address type and its fields?

Reply:
Yes, I'll translate it to English :)

Reply:
Fixed.